Source/WebCore/Modules/url-pattern/URLPatternTokenizer.cpp

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "URLPatternTokenizer.h"
+
+#import <unicode/utf16.h>
+#import <unicode/utf8.h>
+
+namespace WebCore {
+namespace URLPatternUtilities {
+
+// https://urlpattern.spec.whatwg.org/#is-a-valid-name-code-point
+static bool isValidNameCodepoint(UChar codepoint, bool first)
+{
+    if (first)
+        return u_isIDStart(codepoint);
+
+    return u_isIDPart(codepoint);
+}
+
+bool Token::isNull() const
+{
+    return type == TokenType::Open && !index && !value;
+}
+
+// https://urlpattern.spec.whatwg.org/#get-the-next-code-point
+void Tokenizer::getNextCodePoint()
+{
+    if (m_input.is8Bit())
```

Comment:
You'll need to do something like this to fix the build:
```
if (m_input.is8Bit()) {
    auto characters = m_input.span8();
    U8_NEXT_OR_FFFD(characters, m_nextIndex, m_input.length(), m_codepoint);
} else {
    auto characters = m_input.span16();
    U16_NEXT_OR_FFFD(characters, m_nextIndex, m_input.length(), m_codepoint);
}
```

Reply:
The local variable makes the unused variable error go away.

Reply:
I hit this previously here: https://github.com/WebKit/WebKit/pull/27831/files#r1581848413